packages/kbn-typed-react-router-config/src/types/index.ts

```diff
@@ -146,7 +146,7 @@ export interface Router<TRoutes extends RouteMap> {
     path: TPath,
     ...args: TypeAsArgs<TypeOf<TRoutes, TPath, false>>
   ): string;
-  getRoutePath(route: Route): string;
+  getRoutePath(route: RouteWithPath): string;
```

Comment:
I had one minor housekeeping comment about the changes in `packages/kbn-typed-react-router-config`. I'm not sure how long `main+profiler` will diverge from the mainline, but it could useful to move these updates into `main` after this PR to minimize merging issues in the future.